File: lab_02/src/main.py
import time
def compute(x):
    return x*x- x*x + x*4- x*5 + x + x
def main():
    while True:
        inp = input("Enter number of iterations (or non-number to exit): ")
        if inp.isdigit():
            n = int(inp)
        else:
            print("Non-numeric input. Exiting.")
            break
        x = 1.234
        start = time.perf_counter()
        for _ in range(n):
            compute(x)
        end = time.perf_counter()
        print(f"Time for {n} iterations: {end- start:.6f} seconds")
if __name__ == "__main__":
    main()

File: lab_02/src/main.cpp
#include <iostream>
#include <chrono>
#include <string>
#include <cctype>
double compute(double x) {
    return x*x- x*x + x*4- x*5 + x + x;
}
int main() {
    std::string input;
    while (true) {
    std::cout << "Enter number of iterations (or non-number to exit): ";
    std::cin >> input;
    bool is_number = true;
    for (char c : input) {
        if (!std::isdigit(c)) {
        is_number = false;
        break;
        }
    }
    if (!is_number) {
        std::cout << "Non-numeric input. Exiting." << std::endl;
        break;
    }
    long long n = std::stoll(input);
    volatile double x = 1.234;
    auto start = std::chrono::high_resolution_clock::now();
    for (long long i = 0; i < n; ++i) {
        compute(x);
    }
    auto end = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> elapsed = end- start;
    std::cout << "Time for " << n << " iterations: " << elapsed.count() <<
    "seconds" << std::endl;
    }
return 0;
}